tests/trainers/java/files_and_directories.java

```diff
@@ -0,0 +1,216 @@
+/**
+Test to check the Java File module.
+It consists on:
+- Text and binary file read and write
+- File operations (e.g. exists, permissions, rename and delete)
+- Directory operations (e.g. read content, creation)
+*/
+import java.io.*;
+import java.io.IOException;
+import java.util.UUID;
+
+class Test {
+    int step = 1;
+    String textContent = "Content of the text file";
+    int errors = 0;
+
+    private String createTextFile(String filepath) {
+        try {
+            FileWriter file = new FileWriter(filepath);
+            file.write(this.textContent);
+            file.close();
+            return "OK";
+        } catch (IOException ex) {
+            return "KO - " + ex.getMessage();
+        }
+    }
+
+    private String readTextFile(String filepath) {
+        FileReader file = null;
+        try {
+            file = new FileReader(filepath);
+            char [] a = new char[1024];
+            int length = file.read(a);
+            String output = "";
+            for(int i=0; i<length; i++)
+                output = output + a[i];
+            file.close();
+
+            if (output.equals(this.textContent))
+                return "OK";
+            else
+                return "KO - file content doesn't match with '" + this.textContent + "'";
+        } catch (IOException ex) {
+            return "KO - " + ex.getMessage();
+        } finally {
+            if (file != null) {
+                try {
+                    file.close();
+                }catch(IOException ex2) {}
+            }
+        }
+    }
+
+    private String createBinaryFile(String inputFile, String outputFile) {
+        InputStream inputStream = null;
+        OutputStream outputStream = null;
+        try {
+            inputStream = new FileInputStream(inputFile);
+        } catch (IOException ex) {
+            return "KO - " + ex.getMessage();
+        }
+
+        try {
+            outputStream = new FileOutputStream(outputFile);
+        } catch (IOException ex) {
+            try {
+                inputStream.close();
+            }catch(IOException ex2) {}
+            return "KO - " + ex.getMessage();
+        }
+
+        try {
+            int byteRead = -1;
+            while ((byteRead = inputStream.read()) != -1) {
+                outputStream.write((byte)byteRead);
+            }
+            return "OK";
+        } catch (IOException ex) {
+            return "KO - " + ex.getMessage();
+        }finally {
+            try {
+                inputStream.close();
+            }catch(IOException ex2) {}
+            try {
+                outputStream.close();
+            }catch(IOException ex2) {}
+            
+        }
+    }
+
+    private String length(String filepath, int minSize){
+        return new File(filepath).length() == minSize ? "OK" : "KO - file " + filepath + " should have " + minSize + " bytes";
```

Comment:
Wouldn't it be better to use a test framework for this? You're really just re-implementing `assert` here.

Reply:
I am adding assert when it's possible